deps/fast_float/fast_float_strtod.cpp

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2024, Valkey contributors
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *   * Redistributions of source code must retain the above copyright notice,
+ *     this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *   * Neither the name of Redis nor the names of its contributors may be used
+ *     to endorse or promote products derived from this software without
+ *     specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "fast_float.h"
+#include "fast_float_strtod.h"
+#include <cerrno>
+
+extern "C"
+{
+    const char* fast_float_strtod(const char *str, double *value)
+    {
+        double temp = 0; 
+        auto answer = fast_float::from_chars(str, str + strlen(str), temp);
+        *value = temp; 
+        if (answer.ec != std::errc()) { 
+            errno = (answer.ec == std::errc::result_out_of_range) ? ERANGE : EINVAL;
+        }
+        return answer.ptr;
+    }
```

Comment:
```suggestion
    double fast_float_strtod(const char *str, const char *endptr)
    {
        double temp = 0; 
        auto answer = fast_float::from_chars(str, str + strlen(str), temp);
        *endptr = answer.ptr; 
        if (answer.ec != std::errc()) { 
            errno = (answer.ec == std::errc::result_out_of_range) ? ERANGE : EINVAL;
        }
        return temp;
    }
```
Is there a reason we aren't emulating strtod here? It seems odd we are just returning the parameters slightly differently.

Reply:
There is one less if/else, as endptr can be null. This approach removes that if branch. It also removes the requirement to pass NULL since endptr is optional parameter.

Reply:
Then we shouldn't call it strtod, since it's not a strtod replacement.

Reply:
@madolson I agree.

Why can't we return the double though? And is checking for NULL noticeable performance-wise?

In C, `strtod` is the standard way to convert a string to a double, so I think we should prefer emulating it. In C++ there are other standard functions that fast float is emulating.

@parthpatel can you show in a comment how properly emulating strtod would look like? One more if-else you say...